File: AlphaZ-Core/src/main/java/com/alphaz/core/authorization/pojo/entity/OauthApprovalsEntity.java
package com.alphaz.core.authorization.pojo.entity;

import javax.persistence.*;
import java.sql.Timestamp;
import java.util.Objects;

/**
 * @Author: c0der
 * @Date: 下午1:26 2018/2/22
 * @Description:
 */
@Entity
@Table(name = "oauth_approvals", schema = "alphaz", catalog = "")
public class OauthApprovalsEntity {
    private String userId;
    private String clientId;
    private String scope;
    private String status;
    private Timestamp expiresAt;
    private Timestamp lastModifiedAt;
    private Long id;

    @javax.persistence.Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    @Basic
    @Column(name = "userId")
    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    @Basic
    @Column(name = "clientId")
    public String getClientId() {
        return clientId;
    }

    public void setClientId(String clientId) {
        this.clientId = clientId;
    }

    @Basic
    @Column(name = "scope")
    public String getScope() {
        return scope;
    }

    public void setScope(String scope) {
        this.scope = scope;
    }

    @Basic
    @Column(name = "status")
    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    @Basic
    @Column(name = "expiresAt")
    public Timestamp getExpiresAt() {
        return expiresAt;
    }

    public void setExpiresAt(Timestamp expiresAt) {
        this.expiresAt = expiresAt;
    }

    @Basic
    @Column(name = "lastModifiedAt")
    public Timestamp getLastModifiedAt() {
        return lastModifiedAt;
    }

    public void setLastModifiedAt(Timestamp lastModifiedAt) {
        this.lastModifiedAt = lastModifiedAt;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        OauthApprovalsEntity that = (OauthApprovalsEntity) o;
        return Objects.equals(userId, that.userId) &&
                Objects.equals(clientId, that.clientId) &&
                Objects.equals(scope, that.scope) &&
                Objects.equals(status, that.status) &&
                Objects.equals(expiresAt, that.expiresAt) &&
                Objects.equals(lastModifiedAt, that.lastModifiedAt);
    }

    @Override
    public int hashCode() {

        return Objects.hash(userId, clientId, scope, status, expiresAt, lastModifiedAt);
    }
}

File: Core/src/main/kotlin/com/alphaz/core/authorization/auth2/OauthApprovalsEntity.kt
package com.alphaz.core.authorization.auth2

import javax.persistence.*
import java.sql.Timestamp
import java.util.Objects

/**
 * @Author: c0der
 * @Date: 下午1:26 2018/2/22
 * @Description:
 */
@Entity
@Table(name = "oauth_approvals", schema = "", catalog = "")
class OauthApprovalsEntity {
    @get:Basic
    @get:Column(name = "userId")
    var userId: String? = null
    @get:Basic
    @get:Column(name = "clientId")
    var clientId: String? = null
    @get:Basic
    @get:Column(name = "scope")
    var scope: String? = null
    @get:Basic
    @get:Column(name = "status")
    var status: String? = null
    @get:Basic
    @get:Column(name = "expiresAt")
    var expiresAt: Timestamp? = null
    @get:Basic
    @get:Column(name = "lastModifiedAt")
    var lastModifiedAt: Timestamp? = null
    @get:javax.persistence.Id
    @get:GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

}
